### ISSUE TITLE
 resource of type 'microsoft.graph.directoryRole' was found in a resource set that otherwise has entries of type 'microsoft.graph.group'. In OData, all entries in a resource set must have a common base type.

### ISSUE THREAD

**Issue by davidkarlsen** [microsoftgraph/msgraph-sdk-java#695]
### Expected behavior

listtransitivegroups should work fine

### Actual behavior

com.microsoft.graph.http.GraphFatalServiceException: Unexpected exception returned from the service.Error code: Unable to parse error response message

```
  "error": {
    "code": "InternalServerError",
    "message": "A resource of type 'microsoft.graph.directoryRole' was found in a resource set that otherwise has entries of type 'microsoft.graph.group'. In OData, all entries in a resource set must have a common base type.",
    "innerError": {
      "date": "2021-03-12T23:22:11",
      "request-id": "bef202d4-d7b0-4426-8e59-9fea45aa0cf0",
      "client-request-id": "1845763c-7268-494c-a62f-e94f4276e1c3"
    }
  }
}
```

### Steps to reproduce the behavior

happens randomly

 
[AB#8586](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/8586)

**Comment by baywet** [comment]
Hi @davidkarlsen 
Thanks for reaching out.
This looks a lot like the issue you already reported which is a service issue https://github.com/microsoftgraph/msgraph-sdk-serviceissues/issues/37

Can you provide the code sample that lead you to this error please?

**Comment by davidkarlsen** [comment]
core: 1.0.9
sdk: 2.10.0

code:
```
public Optional<AdUser> findByEmail(@NotEmpty @Email String email) {
    var iUserCollectionPage =
        graphServiceClient
            .users()
            .buildRequest()
            .filter(
                String.format(
                    "startsWith(mail,'%s') OR startsWith(userPrincipalName, '%s') OR proxyAddresses/any(x:startsWith(x, 'smtp:%s')) OR otherMails/any(x:startsWith(x, '%s'))",
                    email, email, email, email))
            // userType,preferredName
            .select(
                "otherMails,accountEnabled,employeeId,displayName,givenName,mail,mailNickname,mobilePhone,otherMails,proxyAddresses,surname,userPrincipalName,id")
            .get();
    logger.trace("{}", iUserCollectionPage);
    var optionalUser = iUserCollectionPage.getCurrentPage().stream().findAny();

    if (optionalUser.isEmpty()) {
      return Optional.empty();
    } else {
      Collection<Group> groups;
      try {
// this is the problematic call:
        groups =
            graphServiceClient
                .users(optionalUser.get().id)
                .transitiveMemberOfAsGroup()
                .buildRequest(Arrays.asList(new HeaderOption("ConsistencyLevel", "eventual")))
                .get()
                .getCurrentPage();
      } catch (Exception e) {
        logger.error("Error fetcing groups for [{}] - returning empty set", email, e);
        groups = Collections.emptySet();
      }
      return Optional.of(new AdUser(optionalUser.get(), groups));
    }
  }
```

**Comment by baywet** [comment]
Thanks for the additional details, closing as duplicate of https://github.com/microsoftgraph/msgraph-sdk-serviceissues/issues/37 a service issue is already open for that and we're waiting on the service team to fix that issue.
